cmd/kubeadm/app/cmd/init.go

```diff
@@ -227,17 +226,21 @@ func (i *Init) Run(out io.Writer) error {
 		return err
 	}
 
-	// Is deployment type self-hosted?
+	// PHASE 4: Is this install self-hosted?
 	if i.cfg.SelfHosted {
-		// Temporary control plane is up, now we create our self hosted control
+		// 4a. Temporary control plane is up, now we create our self hosted control
 		// plane components and remove the static manifests:
 		fmt.Println("[self-hosted] Creating self-hosted control plane...")
 		if err := kubemaster.CreateSelfHostedControlPlane(i.cfg, client); err != nil {
 			return err
 		}
+		// 4b. Create the etcd-cluster TPR
```

Comment:
After CreateSelfHostedControlPlane() before CreateEtcdCluster(), it is better to wait TPR API created.
Operator will create TPR API automatically, but it's async.